src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/factory/CaseNoteFactory.kt

```diff
@@ -9,27 +9,37 @@ import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.randomStringUpperCa
 import java.time.LocalDateTime
 
 class CaseNoteFactory : Factory<CaseNote> {
-  private var caseNoteId: Yielded<Long> = { randomInt(0, 1000).toLong() }
-  private var bookingId: Yielded<Long> = { randomInt(0, 1000).toLong() }
+  private var caseNoteId: Yielded<String> = { randomStringUpperCase(6) }
+  private var offenderIdentifier: Yielded<String> = { randomStringUpperCase(8) }
+  private var sensitive: Yielded<Boolean> = { false }
   private var type: Yielded<String> = { "ACP" }
   private var typeDescription: Yielded<String?> = { "Accredited Programme" }
   private var subType: Yielded<String> = { "ASSESSMENT" }
   private var subTypeDescription: Yielded<String?> = { "Assessment" }
   private var source: Yielded<String?> = { null }
   private var creationDateTime: Yielded<LocalDateTime> = { LocalDateTime.now().randomDateTimeBefore(50) }
   private var occurrenceDateTime: Yielded<LocalDateTime> = { LocalDateTime.now().randomDateTimeBefore(50) }
-  private var staffId: Yielded<Long> = { randomInt(0, 1000).toLong() }
+  private var authorUserId: Yielded<String> = { randomInt(0, 1000).toLong().toString() }
   private var authorName: Yielded<String> = { randomStringUpperCase(10) }
   private var text: Yielded<String> = { randomStringUpperCase(20) }
   private var originalTextNote: Yielded<String> = { randomStringUpperCase(20) }
-  private var agencyId: Yielded<String?> = { randomStringUpperCase(5) }
+  private var eventId: Yielded<Int> = { randomInt(0, 1000) }
+  private var locationId: Yielded<String?> = { null }
 
-  fun withCaseNoteId(caseNoteId: Long) = apply {
+  fun withCaseNoteId(caseNoteId: String) = apply {
     this.caseNoteId = { caseNoteId }
   }
 
-  fun withBookingId(bookingId: Long) = apply {
-    this.bookingId = { bookingId }
+  fun withOffenderIdentifier(offenderIdentifier: String) = apply {
+    this.offenderIdentifier = { offenderIdentifier }
+  }
+
+  fun withSensitive(sensitive: Boolean) = apply {
```

Comment:
all the `sensitive` case notes I've encountered are 

```
"type": "OMIC",
 "typeDescription": "OMiC",
```

and most have:

```
"subType": "COMM",
"subTypeDescription": "Communication",
```

so it might be helpful to make the `withSensitive` trait apply those realistic values?